src/components/DepositWidget/index.tsx

```diff
@@ -309,14 +310,50 @@ const BalancesDisplay: React.FC<BalanceDisplayProps> = ({
 }) => {
   const windowSpecs = useWindowSizes()
 
+  const [search, setSearch] = useState('')
+  const [hideZeroBalances, setHideZeroBalances] = useState(false)
+
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>): void => setSearch(e.target.value)
+
+  const handleHideZeroBalances = (e: React.ChangeEvent<HTMLInputElement>): void => setHideZeroBalances(e.target.checked)
+
+  const debouncedSearch = useDebounce(search, 500)
+
+  const filteredBalances = useMemo(() => {
+    if (!debouncedSearch || !balances || balances.length === 0) return balances
+
+    const searchTxt = debouncedSearch.toLowerCase()
+
+    return balances.filter(({ symbol, name, address }) => {
+      return (
+        symbol?.toLowerCase().includes(searchTxt) ||
+        name?.toLowerCase().includes(searchTxt) ||
+        address.toLowerCase().includes(searchTxt)
+      )
+    })
+  }, [debouncedSearch, balances])
+
+  const displayedBalances = useMemo(() => {
+    if (!hideZeroBalances || !filteredBalances || filteredBalances.length === 0) return filteredBalances
+
+    return filteredBalances.filter(({ exchangeBalance, pendingWithdraw, walletBalance }) => {
```

Comment:
Use `totalExchangeBalance` instead to be inline with displayed balances for exchange.